themes/bootstrap3/templates/Recommend/SideFacets.phtml

```diff
@@ -27,24 +27,25 @@
   }
 ?>
 <button class="close-offcanvas btn btn-link" data-toggle="offcanvas"><?=$this->transEsc('navigate_back') ?></button>
-<?php if ($results->getResultTotal() > 0): ?>
-  <h2><?=$this->transEsc($this->slot('side-facet-caption')->get('Narrow Search')) ?></h2>
-<?php endif; ?>
-<?php $checkboxFilters = $this->recommend->getCheckboxFacetSet(); ?>
-<?php $checkboxesShown = false; ?>
-<?php if (count($checkboxFilters) > 0):
+<?php
+  $checkboxFilters = $this->recommend->getCheckboxFacetSet();
+  $checkboxesShown = false;
+  if (count($checkboxFilters) > 0) {
     foreach ($checkboxFilters as $current) {
       if ($results->getResultTotal() > 0 || $current['alwaysVisible']) {
         $checkboxesShown = true;
         break;
       }
     }
-  ?>
-  <?php if ($checkboxesShown): ?>
-    <div class="checkboxFilter">
-      <?=$this->context($this)->renderInContext('Recommend/SideFacets/checkbox-filters.phtml', ['checkboxFilters' => $checkboxFilters, 'results' => $results]); ?>
-    </div>
-  <?php endif; ?>
+  }
+?>
+<?php if ($results->getResultTotal() > 0 || $checkboxesShown): ?>
```

Comment:
Is this block necessary, or could the `<h2>` be included in the `<?php if ($checkboxesShown): ?>` block below?

Reply:
I think it is necessary, because we want to show the header for non-empty result sets and for empty result sets with "always-visible" checkboxes. If we only display the header for visible checkboxes, then it will be missing in situations where we have regular facets but not checkbox facets. Or am I overlooking something?

Reply:
Oh, right. I forgot the regular facets. :) All good then.